pkg/config/config.go

```diff
@@ -574,7 +574,10 @@ func GetSrcType(ctStr string) (SrcType, error) {
 // GetOSSourcePath corrects the current sourcePath depending on local or binary configuration,
 // if Git has been passed, we simply return the source location from LocalConfig
 // this will get the correct source path whether on Windows, macOS or Linux.
-func (lci *LocalConfigInfo) GetOSSourcePath() (path string, err error) {
+//
+// This function also takes in the current working directory + context directory in order
+// to correctly retrieve WHERE the source is located..
+func (lci *LocalConfigInfo) GetOSSourcePath(componentContext string) (path string, err error) {
```

Comment:
why is componentContext required?
Can we use filepath.Dir(filepath.Dir(lci.Filename)) instead ?
Ref: https://github.com/openshift/odo/blob/1c3081ce6406bb99ba5027cade553895efd4da6a/pkg/config/config.go#L92